### PR TITLE
Guest Update

### PR DESCRIPTION
**Pull request by offiongbassey**
Guest Update for editing of guest's information.

### CODE REVIEW

File: src/helpers/guestValidator.js [hunk 1]
```diff
@@ -33,4 +33,17 @@ export const guest_validator = [
         .withMessage("Please Select your Gender"),
     body()
     .custom(body => checkAllowedFields(body, ['name', 'email', 'phone', 'gender']))
+]
```
Comment (eleezs) [inline src/helpers/guestValidator.js]: You are not validating the fields you want the user to make changes too?

File: src/routes/guest.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-import { createGuest } from "../controllers/guest";
+import { createGuest, updateGuest } from "../controllers/guest";
 import { validationHandler } from "../helpers/validation";
-import { guest_validator } from "../helpers/guestValidator";
+import { guest_update_validator, guest_validator } from "../helpers/guestValidator";
 
 const router = express.Router();
 
 router.post('/create', validationHandler(guest_validator), createGuest);
+router.patch('/update/:guest_id', validationHandler(guest_update_validator), updateGuest);
```
Comment (eleezs) [inline src/routes/guest.js:9]: Why did you use PATCH method instead of PUT?

File: src/helpers/guestValidator.js (hunk 1 above)
Comment (offiongbassey) [reply src/helpers/guestValidator.js]: Okay. I will work on that pls

File: src/routes/guest.js (hunk 2 above)
Comment (offiongbassey) [reply src/routes/guest.js:9]: I have changed it to put but patch updates only the field that is required while put updates the entire fields.

File: src/controllers/guest.js [hunk 3]
```diff
@@ -10,4 +10,15 @@ export const createGuest = async(req, res) => {
         await errorHandler(error);
         return responseHandler(res, 500, false, "Something went wrong, please try again later");
     }
+}
+
+export const updateGuest = async(req, res) => {
+    const id = req.params.guest_id;
+    try {
+        await Model.Guest.update(req.body, { where: { id } });
```
Comment (eleezs) [inline src/controllers/guest.js]: assign this to a variable, and return it in response data

File: src/middlewares/validator.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        .withMessage("Please provide a valid Phone Number")
+        .custom(acceptedPhoneNumber)
+        .custom(existingPhone)
+        .trim()
+        .customSanitizer(formatPhoneNumber),
+    body('gender')
+        .exists()
+        .withMessage("Gender is required")
+        .notEmpty()
+        .withMessage("Please Select your Gender")
```
Comment (eleezs) [inline src/middlewares/validator.js]: Kindly remove every instance of please in your message. Simply write
"Gender can not be empty"

File: src/controllers/guest.js (hunk 3 above)
Comment (offiongbassey) [reply src/controllers/guest.js]: I already did that but it kept on returning [1] instead of the data

File: src/middlewares/validator.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        .withMessage("Guest Id is a number")
+        .notEmpty()
+        .withMessage('Guest Id cannot be empty'),
+    body('name')
+        .optional()
+        .exists()
+        .withMessage("Name is required")
+        .notEmpty()
+        .withMessage("Please provide your Name")
+        .isLength({ min: 5, max: 50 })
```
Comment (eleezs) [inline src/middlewares/validator.js]: The isLength validation is no needed for the name, at least the max part.

File: src/helpers/validation.js [hunk 6]
```diff
@@ -6,16 +6,75 @@ import { responseHandler } from "./responseHandler";
...
+    throw new Error("Please provide a valid Phone Number");
+  }
+  return true;
+}
+
+export const formatPhoneNumber = async (phone) => {
+  return `0${ phone?.slice(-10) }`;
+}
+
+export const updateExistingPhone = async (phone, { req }) => {
```
Comment (eleezs) [inline src/helpers/validation.js]: So, there is another approach to this.
All you need from req is the id in the param. So make the function take a payload {phone, id}

File: src/middlewares/validator.js (hunk 5 above)
Comment (offiongbassey) [reply src/middlewares/validator.js]: Okay. I have modified that.

File: src/middlewares/validator.js (hunk 4 above)
Comment (offiongbassey) [reply src/middlewares/validator.js]: Okay Boss. I have removed it.

File: src/helpers/validation.js (hunk 6 above)
Comment (offiongbassey) [reply src/helpers/validation.js]: Ohhh. I see, I have done it.

**Pull request merged**